### PR TITLE
Use apptest library and refactor tests

### PR DESCRIPTION
**Pull request by tomahawk28**
Towards giantswarm/giantswarm#13206

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,27 +3,35 @@ module github.com/giantswarm/certctl/v2
 go 1.14
 
 require (
-	github.com/giantswarm/apprclient/v2 v2.0.0
+	github.com/asaskevich/govalidator v0.0.0-20200108200545-475eaeb16496 // indirect
+	github.com/giantswarm/appcatalog v0.3.2-0.20201202131113-2b8cc1852ce2
```
Comment (tomahawk28) [inline go.mod:7]: TODO: revert it after merge https://github.com/giantswarm/appcatalog/pull/46

File: integration/setup/setup.go [hunk 2]
```diff
@@ -32,12 +38,6 @@ func WrapTestMain(c Config, m *testing.M) {
 		v = m.Run()
 	}
 
-	err = teardown(c)
```
Comment (kopiczko) [inline integration/setup/setup.go:35]: For my curiosity: do we do any teardown at all or we decided we abandon the cluster after the test so we don't need it?

**Review by kopiczko — APPROVED** [review]
That looks way better

File: integration/setup/setup.go (hunk 2 above)
Comment (tomahawk28) [reply integration/setup/setup.go:35]: I think circleCI would tear down a KIND cluster anyway :)

Comment (rossf7) [reply integration/setup/setup.go:35]: Yes for these app tests there are no external resources.

Circle clean up their VMs and when testing locally I usually delete the KIND cluster to have a clean starting state.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**